### PR TITLE
Feat/v1/chart upgrade

### PR DESCRIPTION
**Pull request by envin3**

### CODE REVIEW

File: src/hooks/use-sentinels-historical-data.js [hunk 1]
```diff
@@ -1,25 +1,64 @@
...
-            Object.values(fees).reduce((_acc, { amount_usd }) => {
-              _acc += amount_usd
-              return _acc
-            }, 0)
+        const { data: rawData } = await axios.get('https://pnetwork.watch/static/epoch-dash.json')
+        const { epochs: data } =
+          typeof rawData === 'object'
+            ? rawData
+            : typeof rawData === 'string'
+            ? JSON.parse(rawData.replace(/: ,/g, ': "NaN",').replace(/: }/g, ': "NaN"}'))
```
Comment (oliviera9) [inline src/hooks/use-sentinels-historical-data.js]: Factor out this into a function

File: src/hooks/use-sentinels-historical-data.js [hunk 2]
```diff
@@ -1,25 +1,64 @@
...
-        setEpochs(Object.keys(epochs))
-        setNumberOfNodes(Object.values(epochs).map(({ nodes_count }) => nodes_count))
-        setAccruedFees(
-          Object.values(epochs).map(({ fees }) =>
-            Object.values(fees).reduce((_acc, { amount_usd }) => {
-              _acc += amount_usd
-              return _acc
-            }, 0)
+        const { data: rawData } = await axios.get('https://pnetwork.watch/static/epoch-dash.json')
+        const { epochs: data } =
```
Comment (oliviera9) [inline src/hooks/use-sentinels-historical-data.js]: Factor out this as well. getEpochsFromRawData()

**Review by oliviera9 — COMMENTED** [review]
I would factor out those large blocks.
And add unit tests as well.

File: src/hooks/use-sentinels-historical-data.js [hunk 3]
```diff
@@ -1,25 +1,64 @@
...
-        } = await axios.get('https://pnetwork.watch/api/datasources/proxy/3')
-        setEpochs(Object.keys(epochs))
-        setNumberOfNodes(Object.values(epochs).map(({ nodes_count }) => nodes_count))
-        setAccruedFees(
-          Object.values(epochs).map(({ fees }) =>
-            Object.values(fees).reduce((_acc, { amount_usd }) => {
-              _acc += amount_usd
-              return _acc
-            }, 0)
+        const { data: rawData } = await axios.get('https://pnetwork.watch/static/epoch-dash.json')
```
Comment (oliviera9) [inline src/hooks/use-sentinels-historical-data.js]:
```suggestion
        const { data: rawData } = await axios.get(EPOCHS_ENDPOINT)
```

File: src/hooks/use-sentinels-historical-data.js [hunk 4]
```diff
@@ -1,25 +1,66 @@
...
   const [numberOfNodes, setNumberOfNodes] = useState([])
   const [epochs, setEpochs] = useState([])
-  const [accruedFees, setAccruedFees] = useState([])
+  const [accruedNodesFees, setAccruedNodesFees] = useState([])
+
+  const handlePartialData = (data) =>
+    typeof data === 'object'
+      ? data
+      : typeof data === 'string'
+      ? JSON.parse(data.replace(/: ,/g, ': "NaN",').replace(/: }/g, ': "NaN"}'))
```
Comment (oliviera9) [inline src/hooks/use-sentinels-historical-data.js]: Put a comment explaining why this processing is needed as it's not immediate. Would you remember it in a year?

File: src/hooks/__tests__/use_sentinels-historical-data.test.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import { handlePartialData } from '../use-sentinels-historical-data'
+import { completeData, partialData } from './api.data'
+
+it('elaborates data', () => {
```
Comment (oliviera9) [inline src/hooks/__tests__/use_sentinels-historical-data.test.js]:
```suggestion
it('should elaborate data', () => {
```

**Review by oliviera9 — APPROVED** [review]
Maybe add `should` to tests. LGTM though

**Comment by envin3** [comment]
> Maybe add `should` to tests. LGTM though

yes right

**Pull request merged**